source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -1626,7 +1633,7 @@ void COpenGLDriver::drawMeshBuffer(const IGPUMeshBuffer* mb)
     if (!found->nextState.pipeline.graphics.pipeline)
         return;
 
-    found->updateNextState_vertexInput(mb->getVertexBufferBindings(), mb->getIndexBufferBinding().buffer.get(), nullptr, nullptr);
+    found->updateNextState_vertexInput(mb->getVertexBufferBindings(), mb->getIndexBufferBinding().buffer.get(), found->nextState.vertexInputParams.indirectDrawBuf.get(), found->nextState.vertexInputParams.parameterBuf.get());
```

Comment:
I commented this already in the other PR #411 , `drawMesh` is not for indirect draws.

Reply:
yea, thats why im not modifying indirect draw buffer and parameter buffer state. Passing nullptr as 2 last args to this functions results in setting indirect draw and parameter buffer bindings to 0 for really no reason

Reply:
setting them to 0 in a `drawMesh` is the correct solution... otherwise you have state pollution! (which is exactly why we invented the whole state tracking system in the first place).

Imagine firing off a few indirect draws before a `drawMesh` and then getting random results from the `drawMesh` depending on which was the last indirect draw to go before it!

Reply:
i dont understand how indirect draw buffer binding affects direct draw calls, but ok i'll revert this

Reply:
ah fuck, you were right (I forgot that non indirect does not get affected)... revert the revert